### PR TITLE
Replace dependency on /etc/lsb-release with /usr/bin/lsb_release to permit Debian compatibility - #1990268

### PR DESCRIPTION
**Pull request by Borgquite**
As requested in Bug #1990268 here is some code that should replace the /etc/lsb-release dependency (which is Ubuntu-specific) with more generic /usr/bin/lsb_release dependency, allowing package reporting to work on Debian.

Looking through the code there are various references in the tests - I'm afraid this is beyond my skills, but hoping someone with more knowledge might be able to help out here!

### CODE REVIEW

**Comment by Perfect5th** [comment]
The tests need to pass for the PR to pass our CI checks.

If you've allowed edits by maintainers on this PR, I'll fix the tests.

**Comment by Perfect5th** [comment]
@Borgquite I'd appreciate it if you'd to test my latest changes on debian.

File: landscape/lib/lsb_release.py
Comment (silverdrake11) [inline landscape/lib/lsb_release.py:26]: I tried this and I am getting FileNotFound error instead of CalledProcessError if lsb-release is not installed on the system so it doesn't fall back to reading the file

Comment (Perfect5th) [reply landscape/lib/lsb_release.py:26]: Right. overlooked that, will fix.

Comment (silverdrake11) [reply landscape/lib/lsb_release.py]: I think I would prefer to using one method, probably the command line. It keeps it simple to use one method and also in debian falling back to the file is not going to work, (unless you use the /etc/os-release but then it's gonna be different keys for it). lsb-release should be installed in ubuntu and I'm fine with the release information missing from server if this is not installed, but I can also understand having it be foolproof in ubuntu  so your call

**Review by silverdrake11 — CHANGES_REQUESTED** [review]

**Comment by Perfect5th** [comment]
I'm going to respectfully decline restricting it to one method, but not only for "foolproofness". Currently a lot of the application code passes in `lsb_release_filename`, so not having it handle the file-reading case necessitates a _lot_ of additional touches to the code-base.

File: landscape/lib/lsb_release.py
Comment (Perfect5th) [reply landscape/lib/lsb_release.py]: I'm going to respectfully decline restricting it to one method, but not only for "foolproofness". Currently a lot of the application code passes in `lsb_release_filename`, so not having it handle the file-reading case necessitates a _lot_ of additional touches to the code-base.

**Review by silverdrake11 — APPROVED** [review]

**Comment by Borgquite** [comment]
@Perfect5th Tested on Debian 10. It didn't work because of the dependence on DEVNULL, which is only in Python > 3.3 (Debian 10 uses Python 2.7.16
Reverted to os.devnull, seems to work OK now?

**Comment by Borgquite** [comment]
Great! Will look forward to this being released :)

**Pull request merged**
